management/server/account.go

```diff
@@ -225,6 +239,53 @@ func (am *DefaultAccountManager) updateIDPMetadata(userId, accountID string) err
 	return nil
 }
 
+func mergeLocalAndQueryUser(queried idp.UserData, local User) *UserInfo {
+	return &UserInfo{
+		ID:    queried.ID,
+		Email: queried.Email,
+		Name:  queried.Name,
+		Role:  string(local.Role),
+	}
+}
+
+// batched request for users from auth0
+func (am *DefaultAccountManager) GetUsersFromAccount(accountID string) ([]*UserInfo, error) {
+	account, err := am.GetAccountById(accountID)
+	if err != nil {
+		return nil, err
+	}
+
+	queriedUsers := make([]*idp.UserData, 0)
+
+	// https://auth0.com/docs/manage-users/user-search/retrieve-users-with-get-users-endpoint#limitations
+	// auth0 limitation of 1000 users via this endpoint
+	for page := 0; page < 20; page++ {
+		batch, err := am.idpManager.GetBatchedUserData(accountID, page)
```

Comment:
You should handle the case where there are no IDP manager credentials (self-hosted) where we usually don't call idp manager methods and in this case, `queriedUsers` will always be empty.

Reply:
Good point @mlsmaycon 
I also think we need a test for this